core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -469,15 +481,15 @@ default Executor makeContextAware(Executor executor) {
      * sure to propagate the current {@link RequestContext} into the callback execution.
      */
     default ExecutorService makeContextAware(ExecutorService executor) {
```

Comment:
Don't we need to check whether the specified `executor` is `ContextAwareExecutorService` like you did in `ContextAwareScheduledExecutorService.of`?